cmd/executor/cmd/root.go

```diff
@@ -110,21 +113,40 @@ func checkDockerfilePath() error {
 	return errors.New("please provide a valid path to a Dockerfile within the build context")
 }
 
-// resolveSourceContext unpacks the source context if it is a tar in a GCS bucket
+// resolveSourceContext unpacks the source context if it is a tar in a bucket
 // it resets srcContext to be the path to the unpacked build context within the image
 func resolveSourceContext() error {
 	if srcContext == "" && bucket == "" {
-		return errors.New("please specify a path to the build context with the --context flag or a GCS bucket with the --bucket flag")
+		return errors.New("please specify a path to the build context with the --context flag or a bucket with the --bucket flag")
 	}
-	if bucket == "" {
-		return nil
-	}
-	logrus.Infof("Using GCS bucket %s as source context", bucket)
+
 	buildContextPath := constants.BuildContextDir
-	if err := util.UnpackTarFromGCSBucket(bucket, buildContextPath); err != nil {
+	var contextExecutor buildcontext.BuildContext
+
+	// if no prefix use Google Cloud Storage as default for backwards compability
+	if !strings.Contains(srcContext, "://") {
+		if bucket != "" {
+			srcContext = fmt.Sprintf("gs://%s", bucket)
+		} else if bucket == "" {
+			srcContext = fmt.Sprintf("dir://%s", srcContext)
+		}
+	}
+
+	// if no context is set, add default file context.tar.gz
+	if !strings.HasSuffix(srcContext, ".tar.gz") {
+		srcContext += "/" + constants.ContextTar
+	}
+
+	contextExecutor, err := buildcontext.GetBuildContext(srcContext)
```

Comment:
If a user specifies the --bucket flag instead of --srcContext, how will that get passed in here?

Reply:
The idea is the following: 
If you specify —context that goes with priority. 
If you don’t put a prefix in there or it’s not specified: 
If bucket is set: use the bucket value with gcs. 
Bucket is empty but no prefix is set: use the context value as directory. 

That way we keep it backwards compatible.